### PR TITLE
BEI-15: Log in users

### PR DESCRIPTION
**Pull request by RICHYDOS**
WHY THIS PR?
* To give users the ability to log in to their account on foodies Haven

WHAT DOES THIS PR DO?
* Sets up the User Login Route

HOW CAN THIS BE TESTED
* Starting the Server:
```
npm run start:dev
```
* After registering, go to the following route: 
  "http://localhost:PORT/api/users/login" 
 where PORT stands in for your port number.

* Typing in the req body:
```
"email":"testing@gmail.com",
"password":"testing123"
```
Feel free to use whichever details you registered with

THINGS YOU MAY WANT THE REVIEWER TO NOTE
N/A

### CODE REVIEW

**Review by hizikex — APPROVED** [review]

File: src/core/controllers/user.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      const result: UserLoginResponse = {
+        email: user.email,
+        token: accessToken,
+        username: user.username,
+        bio: user.bio,
+        image: user.image
+      };
+      return result;
+    } else {
+      throw new Error('Error on our side, sorry for this');
```
Comment (LucianaDera) [inline src/core/controllers/user.ts:62]: can we change the error message... there was an error while loggin, please try again.

File: src/core/controllers/user.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+};
+
+export const processLogin = async (body: UserLoginRequest): Promise<UserLoginResponse> => {
+
+  const userRepository = new UserRepository();
+  const user = await userRepository.findOne({ where: { email: body.email } });
+  if (user) {
+    const hashedPassword: string = user.password;
+
+    // Compare client password with db password
```
Comment (LucianaDera) [inline src/core/controllers/user.ts:38]: Any purpose for the comment

File: src/core/controllers/user.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  });
+  
+  return { message: 'User registration successful', user: newUser };
+};
+
+export const processLogin = async (body: UserLoginRequest): Promise<UserLoginResponse> => {
+
+  const userRepository = new UserRepository();
+  const user = await userRepository.findOne({ where: { email: body.email } });
+  if (user) {
```
Comment (LucianaDera) [inline src/core/controllers/user.ts:35]: can we add an error handler if the user doesn't exist
if(!user){
  throw new Error('Invalid email or password.')
}

**Review by LucianaDera — CHANGES_REQUESTED** [review]

File: src/core/controllers/user.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        //Payload
+        {
+          user: {
+            username: user.username
+          }
+        },
+        //Access Token Secret Key
+        setting.secretKey,
+        // Options like token expiry
+        { expiresIn: '4h' },
```
Comment (LucianaDera) [inline src/core/controllers/user.ts:50]: Can include this information in our env file

**Review by LucianaDera — CHANGES_REQUESTED** [review]

File: src/core/controllers/user.ts (hunk 4 above)
Comment (RICHYDOS) [reply src/core/controllers/user.ts:50]: It is in the env file as "ACCESS_TOKEN_KEY". It was imported in the application.ts file

Comment (LucianaDera) [reply src/core/controllers/user.ts:46]: actually referring to the expiration time

**Review by LucianaDera — APPROVED** [review]

**Pull request merged**